include/swift/AST/Expr.h

```diff
@@ -4567,25 +4567,28 @@ class KeyPathExpr : public Expr {
     
     
     llvm::PointerIntPair<Expr *, 3, Kind> SubscriptIndexExprAndKind;
+    ArrayRef<Identifier> SubscriptLabels;
     Type ComponentType;
     SourceLoc Loc;
     
     explicit Component(DeclNameOrRef decl,
                        Expr *indexExpr,
+                       ArrayRef<Identifier> subscriptLabels,
                        Kind kind,
                        Type type,
                        SourceLoc loc)
       : Decl(decl), SubscriptIndexExprAndKind(indexExpr, kind),
+        SubscriptLabels(subscriptLabels),
```

Comment:
Must we rely on the caller to copy the subscript labels into the `ASTContext`? It feels so brittle when we do this.